### PR TITLE
Mo/improve private key description

### PR DESCRIPTION
**Pull request by moAlobaidi**
This PR addresses and closes issue #196.

### CODE REVIEW

File: documentation/concepts/00_accounts.md [hunk 1]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
 1. Sample a 32 byte `seed` from random
     
 2. Construct private key components
-    - `sk_sig` = Blake2s(`seed`, 0)
-    - `sk_prf` = Blake2s(`seed`, 1)
-    - `r_pk` = Blake2s(`seed`, `counter`)
+    - `sk_sig` = BLAKE2s(`seed` ∥ 0), where | denotes concatenation
```
Comment (collinc97) [inline documentation/concepts/00_accounts.md:107]:
```suggestion
    - `sk_sig` = BLAKE2s(`seed` | 0), where | denotes concatenation
```

File: documentation/concepts/00_accounts.md [hunk 2]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
...
+    - `sk_prf` = BLAKE2s(`seed`∥ 1)
```
Comment (collinc97) [inline documentation/concepts/00_accounts.md:108]:
```suggestion
    - `sk_prf` = BLAKE2s(`seed` | 1)
```

File: documentation/concepts/00_accounts.md [hunk 3]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
...
+    - `r_pk` = BLAKE2s(`seed` ∥ `counter`)
```
Comment (collinc97) [inline documentation/concepts/00_accounts.md:109]:
```suggestion
    - `r_pk` = BLAKE2s(`seed` | `counter`)
```

File: documentation/concepts/00_accounts.md [hunk 4]
```diff
@@ -127,9 +131,9 @@ Given global instantiated Aleo parameters and subroutines.
 ```mermaid
 graph TD
 	A["Seed (32 Bytes)"] 
-    A --> |"Blake2s(Seed, 0)"| B(sk_sig)
-    A --> |"Blake2s(Seed, 1)"| C(sk_prf)
-    A --> |"Blake2s(Seed, counter)"| D(r_pk)
+    A --> |"BLAKE2s(Seed, 0)"| B(sk_sig)
+    A --> |"BLAKE2s(Seed, 1)"| C(sk_prf)
+    A --> |"BLAKE2s(Seed, counter)"| D(r_pk)
```
Comment (collinc97) [inline documentation/concepts/00_accounts.md:136]:
```suggestion
    A --> |"BLAKE2s(Seed, 0)" | B(sk_sig)
    A --> |"BLAKE2s(Seed, 1)" | C(sk_prf)
    A --> |"BLAKE2s(Seed, counter)" | D(r_pk)
```

**Review by collinc97 — CHANGES_REQUESTED** [review]
The ∥ shows up strangely formatted on GitHub we should stick with `|` since it is used in other parts of the docs.

**Comment by moAlobaidi** [comment]
@collinc97 SGTM

File: documentation/concepts/00_accounts.md [hunk 5]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
...
-    - `sk_prf` = Blake2s(`seed`, 1)
-    - `r_pk` = Blake2s(`seed`, `counter`)
+    - `sk_sig` = BLAKE2s(`seed` | 0), where | denotes concatenation
+    - `sk_prf` = BLAKE2s(`seed` | 1)
+    - `r_pk` = BLAKE2s(`seed` | `counter`)
            
 3.`private_key` = (`seed`, `sk_sig`, `sk_prf`, `r_pk`)
 
-`counter` is a `u16` value that is iterated on until a valid `view_key` can be derived from `private_key`
+The 0 and 1 passed as second arguments to calculate `sk_sig` and `sk_prf` are encoded as unsigned 16-bit integers.
```
Comment (acoglio) [inline documentation/concepts/00_accounts.md:113]: I would replace the line "The 0 and 1 ..." with:

The 0 and 1 used to calculate `sk_sig` and `sk_prf` are each encoded as an unsigned 16-bit integer and turned into two bytes in little endian order before being concatenated to the right of the `seed`, then the resulting byte sequence is passed to BLAKE2s.

File: documentation/concepts/00_accounts.md [hunk 6]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
 1. Sample a 32 byte `seed` from random
     
 2. Construct private key components
-    - `sk_sig` = Blake2s(`seed`, 0)
-    - `sk_prf` = Blake2s(`seed`, 1)
-    - `r_pk` = Blake2s(`seed`, `counter`)
+    - `sk_sig` = BLAKE2s(`seed` | 0), where | denotes concatenation
+    - `sk_prf` = BLAKE2s(`seed` | 1)
+    - `r_pk` = BLAKE2s(`seed` | `counter`)
```
Comment (acoglio) [inline documentation/concepts/00_accounts.md:110]: I would move 'where `|` denotes concatenation right here, in its own line, after the itemized lists.

Comment (acoglio) [reply documentation/concepts/00_accounts.md:110]: I would also add 'and where `BLAKE2s` denotes unkeyed BLAKE2s-256, as defined in [RFC 7693](https://www.rfc-editor.org/rfc/rfc7693).

File: documentation/concepts/00_accounts.md [hunk 7]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
...
            
 3.`private_key` = (`seed`, `sk_sig`, `sk_prf`, `r_pk`)
 
-`counter` is a `u16` value that is iterated on until a valid `view_key` can be derived from `private_key`
+The 0 and 1 passed as second arguments to calculate `sk_sig` and `sk_prf` are encoded as unsigned 16-bit integers.
+`counter` is a `u16` value that is iterated on, starting from 2, until a valid `view_key` can be
```
Comment (acoglio) [inline documentation/concepts/00_accounts.md:114]: valid `view_key` => valid `view_key` (as defined below)

File: documentation/concepts/00_accounts.md [hunk 8]
```diff
@@ -104,13 +104,17 @@ Given global instantiated Aleo parameters and subroutines.
...
+    - `sk_sig` = BLAKE2s(`seed` | 0), where | denotes concatenation
+    - `sk_prf` = BLAKE2s(`seed` | 1)
+    - `r_pk` = BLAKE2s(`seed` | `counter`)
            
 3.`private_key` = (`seed`, `sk_sig`, `sk_prf`, `r_pk`)
 
-`counter` is a `u16` value that is iterated on until a valid `view_key` can be derived from `private_key`
+The 0 and 1 passed as second arguments to calculate `sk_sig` and `sk_prf` are encoded as unsigned 16-bit integers.
+`counter` is a `u16` value that is iterated on, starting from 2, until a valid `view_key` can be 
+derived from `private_key`. The endianness of the 2 bytes that form these u16s are little-endian.
```
Comment (acoglio) [inline documentation/concepts/00_accounts.md:115]: I would replace the sentence 'The endianness...' with 'Similarly to 0 and 1 used to calculate `sk_sig` and `sk_prf`, the unsigned 16-bit integer `counter` is turned into two bytes in little endian order before being concatenated to the right of the `seed`.

**Review by acoglio — CHANGES_REQUESTED** [review]
Thanks! I've added a few suggestions. (Super-picky, I know.)

**Review by acoglio — APPROVED** [review]

**Comment by moAlobaidi** [comment]
Thanks for addressing those changes @collinc97. Will merge now and cite this PR under the original issue before closing it.

**Pull request merged**